tests/functional/views/basic_lti_launch_test.py

```diff
@@ -87,91 +55,31 @@ def test_basic_lti_launch(
-                id="canvas file launch with exiting legacy SpeedGrader row",
-            ),
-            param(
-                "canvas_file_legacy_speedgrader_launch_params",
-                "canvas_file_assignment",
-                id="SpeedGrader launch with existing regular DB row",
-                marks=pytest.mark.xfail(
-                    reason="We don't have information of the real `resource_link_id`"
-                    "and we update the legacy row. The launch will have the right URL thought."
-                    "The wrong row, with the updated document_url, will be used to get the via_url",
-                    strict=True,
-                ),
-            ),
-            param(
-                "canvas_file_launch_params",
-                "canvas_file_assignment",
-                id="canvas file launch with existing DB row",
-            ),
-            param(
-                "canvas_file_legacy_speedgrader_launch_params",
-                None,
-                id="SpeedGrader launch with no existing DB rows",
-            ),
-            param(
-                "canvas_file_launch_params",
-                None,
-                id="canvas file launch with no existing DB rows",
-            ),
-            param(
-                "canvas_file_speedgrader_launch_params",
-                "canvas_file_assignment",
-                id="SpeedGrader canvas file launch with existing DB row",
-            ),
-            param(
-                "canvas_file_speedgrader_launch_params",
-                None,
-                id="SpeedGrader launch with no existing DB rows",
-            ),
-            param(
-                "canvas_file_speedgrader_launch_params",
-                "legacy_speedgrader_assignment",
-                id="SpeedGrader launch with legacy SpeedGrader DB row",
-            ),
-        ],
-    )
-    def test_canvas_file_assignment(
-        self,
-        do_lti_launch,
-        db_session,
-        launch_params_fixture_name,
-        existing_assignment_fixture_name,
-        request,
+    def test_basic_lti_launch_canvas_deep_linking_canvas_file(
+        self, do_lti_launch, db_session, canvas_file_launch_params
     ):
-        if existing_assignment_fixture_name:
-            request.getfixturevalue(existing_assignment_fixture_name)
-
-        get_params, post_params = request.getfixturevalue(launch_params_fixture_name)
-
-        canvas_file_id = post_params["file_id"]
-        canvas_course_id = post_params["custom_canvas_course_id"]
+        get_params, post_params = canvas_file_launch_params
 
         response = do_lti_launch(
             get_params=get_params, post_params=post_params, status=200
         )
 
+        js_config = self.get_client_config(response)
+        assert js_config["mode"] == JSConfig.Mode.BASIC_LTI_LAUNCH
         assert (
-            self.get_client_config(response)["mode"] == JSConfig.Mode.BASIC_LTI_LAUNCH
+            js_config["api"]["viaUrl"]["path"]
```

Comment:
Different test for Canvas file because this is different, we point the FE to our API. Adding an assertion about that.